utils/kernel.c

```diff
@@ -615,10 +615,10 @@ int record_kernel_trace_pipe(struct uftrace_kernel_writer *kernel, int cpu, int
 	if (n < 0) {
 		if (errno == EINTR)
 			goto retry;
-		if (errno == EAGAIN)
+		if ((errno == EAGAIN) || (errno == ENODEV))
```

Comment:
There is no need (and it's not usual) to add the extra braces around `==` because it precedence and order of evaluation comes before the logical or `||`: 
```suggestion
		if (errno == EAGAIN || errno == ENODEV)
```
A detailed reference: https://learn.microsoft.com/en-us/cpp/c-language/precedence-and-order-of-evaluation?view=msvc-170

Otherwise, LGTM.

Reply:
Oh, cool.  I didn't notice this feature.  Yes I prefer no parenthesis.

Reply:
Removed the redundant brackets and force pushed for a simple commit history.

Reply:
It looks the brackets are still there. Please check again.

Reply:
Indeed, sorry + fixed!